### PR TITLE
HUG-219: throw newProgrammer error takes optional error object

### PR DESCRIPTION
**Pull request by OlafurTorfi**
If error object is operational oops error, it keeps its category.

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class Oops extends Error {
 }
 
 export const getErrorCategory = (err): ErrorCategory => {
```
Comment (bergur) [inline lib/index.ts]: should err be optinal parameter now?

File: lib/index.ts [hunk 2]
```diff
@@ -109,11 +109,16 @@ export const operationalOops = (message: string, context?: {}) => {
...
-export const newProgrammerOops = (message: string, context?: {}) => {
+export const newProgrammerOops = (
+    message: string,
+    context?: {},
+    error?: any
+) => {
     return new Oops({
         message,
-        category: constants.ProgrammerError,
+        category: getErrorCategory(error),
```
Comment (bergur) [inline lib/index.ts:119]: or maybe handle it here that sometimes there is no error.
category: error ? getErrorCategory(error) || constants.ProgrammerError

Or change signature on getErrorCategory that error can be optional?

File: lib/index.ts (hunk 1 above)
Comment (OlafurTorfi) [reply lib/index.ts]: Err is any type, so it won't be checked on compile time anyway.

Maybe adding it as optional will make that more clear

File: lib/index.ts (hunk 2 above)
Comment (OlafurTorfi) [reply lib/index.ts:119]: I will change signature on getErrorCategory to show that error can be undefined

**Pull request closed without merging**